packages/vm/src/errors/backtrace.rs

```diff
@@ -0,0 +1,44 @@
+use core::fmt::{Debug, Display, Formatter, Result};
+use std::backtrace::Backtrace;
+
+/// This wraps an actual backtrace to achieve two things:
+/// - being able to fill this with a stub implementation in `no_std` environments
+/// - being able to use this in conjunction with [`thiserror::Error`]
+pub struct BT(Box<Backtrace>);
+
+impl BT {
+    #[track_caller]
+    pub fn capture() -> Self {
+        BT(Box::new(Backtrace::capture()))
```

Comment:
This is cool as it allows us to enable backtraces in libwasmvm at runtime without recompile using env variables, right?

Reply:
Yes, I think that's how it should work. Haven't tested that though.